### ISSUE TITLE
invoke.bat missing. I follow every instruction in the manual

### ISSUE THREAD

**Issue by CoderxASM** [invoke-ai/InvokeAI#1949]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS

Windows

### GPU

cuda

### What happened?

After install invoke.bat is missing. OS I use is windows 11

### Screenshots

![image](https://user-images.githubusercontent.com/120391917/207067208-62f286df-d21b-4836-ba0b-62a2905f4cfa.png)

**Comment by mauwii** [comment]
at which point in the instructions does it fail (asking since this screenshot looks like a repo clone instead of the automated installer folder)

**Comment by mauwii** [comment]
> Same problem here. Just tried a fresh install, updated python and all other somftwere, also tried on a VM but no luck. Invoke.bat is not located in any folder even if installer says everything installed as it should..

When I did the Installation on a Windows Notebook, I had the invoke.bat inside the folder where I installed to ($HOME/InvokeAI/invoke.bat), but not tested the final revision on Windows

**Comment by mauwii** [comment]
- which instructions did you follow?
- which directory are you looking for the bat?

**Comment by chrl** [comment]
Same here. Installed using automated installer, windows 11, looking for invoke.bat in ~./invokeai/ (default installation path)

**Comment by Parkie-95** [comment]
Having the exact same issue here as well no invoke.bat on Windows 11. Do we have a copy of the .bat code that could be pasted in so we can make our own for those of us with this issue in the meantime ?

**Comment by mauwii** [comment]
Sorry - mixed things up in my broken brain.exe - just remember that there was a different  error on my windows 10 machine in the last test (also don't have a old folder anymore)

But the template to the batfile is [here](https://github.com/invoke-ai/InvokeAI/blob/main/installer/templates/invoke.bat.in) ;)

**Comment by FHeifer** [comment]
Here I am,  in the same boat.

**Comment by BryceBarbara** [comment]
I'm finding that it's trying to install to the root of my drive. So for those of you who are also having this issue, do you find invoke.bat at `C:/invoke.bat`?

**Comment by lstein** [comment]
Folks, could somebody see if this version clears up the problem?

[InvokeAI-installer-2.2.4-windows.zip](https://github.com/invoke-ai/InvokeAI/files/10211725/InvokeAI-installer-2.2.4-windows.zip)

**Comment by lstein** [comment]
> Turns out I had python `3.8` installed and it was causing issues. Updating to 3.11, removing the old Python38 folder from the PATH environmental variable, and restarting my terminal fixed it.

The install script didn't complain that version 3.8 is too old? It is supposed to.

**Comment by BryceBarbara** [comment]
@lstein Nope, the new installer you linked gets another error and immediately exits. 

![image](https://user-images.githubusercontent.com/24460210/207147369-1f4e2db4-d02d-452c-bf2a-10dc2d12f40f.png)

The install script didn't complain about 3.8, no.

Side Node: If I was a normal user, I'd have no way of knowing what the error was since double clicking on the .bat file, starting the install, then getting the error causes the terminal window to immediately close because there's no `pause` statement or anything. There might be a way to avoid that.

**Comment by lstein** [comment]
@BryceBarbara 
> I'm finding that it's trying to install to the root of my drive. So for those of you who are also having this issue, do you find invoke.bat at `C:/invoke.bat`?

The script is **supposed** to offer you the option of installing the `invokeai` directory anywhere, including on an external drive. The default location is C:\Users\your-name\invokeai . Before the script installed to C:\

**Comment by BryceBarbara** [comment]
> @BryceBarbara
> 
> > I'm finding that it's trying to install to the root of my drive. So for those of you who are also having this issue, do you find invoke.bat at `C:/invoke.bat`?
> 
> The script is **supposed** to offer you the option of installing the `invokeai` directory anywhere, including on an external drive. The default location is C:\Users\your-name\invokeai . Before the script installed to C:\ , what path did you choose?

Prior to updating to 3.11, it didn't even ask about what path to use. It'd just _do things_ after the initial "Press any key to continue" and then exit.

**Comment by BryceBarbara** [comment]
Yeah, using a .bat file is gonna introduce these sort of issues and I'd recommend against using them. Although it's more complicated, I'd recommend using a powershell script.

**Comment by lstein** [comment]
> Yeah, using a .bat file is gonna introduce these sort of issues and I'd recommend against using it. Although it's more complicated, I'd recommend using a powershell script.

Actually we're rewriting everything as a python script to get portability. But for now the .bat file is what we've got. Unfortunately I can't reproduce the `set` related crash on the Windows machine I have access to (Windows 10 pro, powershell). Would you mind commenting out the @echo off statement at the top of the script and running it again so that I can see which set is triggering the error? Thanks.

**Comment by BryceBarbara** [comment]
<details>
  <summary>Relevant Error Log</summary>

  ```txt
C:\Users\Home-Primary\Downloads\InvokeAI-Installer>set /p selection=Select the path to install InvokeAI's directory into [C:\Users\Home-Primary]:
Select the path to install InvokeAI's directory into [C:\Users\Home-Primary]:
set was unexpected at this time.

C:\Users\Home-Primary\Downloads\InvokeAI-Installer>     if  == "" set selection=C:\Users\Home-Primary
  ```
</details>

**Comment by lstein** [comment]
Thanks. That helped pin down the problem. Please give this version a try

[InvokeAI-installer-2.2.4-windows.zip](https://github.com/invoke-ai/InvokeAI/files/10211950/InvokeAI-installer-2.2.4-windows.zip)

(As Bullwinkle used to say "This time for sure!")

**Comment by lstein** [comment]
> And the install script didn't complain about 3.8, no

H'mmm. I've only tested the version checking part of the code by setting the minimum requirements to something high (like 3.12.1.) and seeing if the version checker complained. What does `python -V` print out when you are using 3.8, or is it gone from your system now?

**Comment by BryceBarbara** [comment]
> Thanks. That helped pin down the problem. Please give this version a try
> 
> [InvokeAI-installer-2.2.4-windows.zip](https://github.com/invoke-ai/InvokeAI/files/10211950/InvokeAI-installer-2.2.4-windows.zip)
> 
> (As Bullwinkle used to say "This time for sure!")

Yeah, that worked.

> > And the install script didn't complain about 3.8, no
> 
> H'mmm. I've only tested the version checking part of the code by setting the minimum requirements to something high (like 3.12.1.) and seeing if the version checker complained. What does `python -V` print out when you are using 3.8, or is it gone from your system now?

Sorry, it's been purged from my system 😢

**Comment by lstein** [comment]
> > Thanks. That helped pin down the problem. Please give this version a try
> > [InvokeAI-installer-2.2.4-windows.zip](https://github.com/invoke-ai/InvokeAI/files/10211950/InvokeAI-installer-2.2.4-windows.zip)
> > (As Bullwinkle used to say "This time for sure!")
> 
> Yeah, that worked.

Just to confirm, since I can hardly believe my ears at this point, the most recent Windows 2.2.4 installer bat script worked and you got a functional install?

**Comment by lstein** [comment]
I've never been happier to close an issue in my life. Thanks all for bearing through this!

**Comment by Dredlox456** [comment]
So you have to uninstall the previous version right? How is it done in Windows? That conda thing doesnt work from a command prompt or am i doing it wrong?
